### ISSUE TITLE
1.15.2 - error fetching recipes - Polymorph dislikes mod updates on existing worlds

### ISSUE THREAD

**Issue by ArMoReDfUrY** [TheIllusiveC4/Polymorph#10]
#### Versions (Be specific, do not write "latest"):
* Polymorph:  polymorph-FORGE-1.15.2-0.3
* Forge: 31.2.24
#### Steps to Reproduce:
1. put rose bush in crafting grid - should be no output
2. 
3.
... it may be some weird conflict but not sure on that, the actual crash came from the silent gear table when i tried to collect the output. i was trying to craft a rose bush with a botania mortar&pestle. the table broke after i left the mortar&pestle in the table and closed the gui to gather the flowers, had to break it to get the mortar back. 

The Logs show it froze due to: [top.theillusivec4.polymorph.Polymorph/]: Attempted to craft using [1 pestle_and_mortar, 1 air, 1 air, 1 air, 1 air, 1 air, 1 air, 1 air, 1 air] but an error occurred while fetching recipes!
 
And crashed due to: [net.minecraft.client.Minecraft/]: Reported exception thrown!
net.minecraft.crash.ReportedException: mouseClicked event handler

Crash Log:  will add it as soon as pastebin stops bugging out...
EDIT: crash log -   https://gist.github.com/ArMoReDfUrY/239b4e738349fc8450611d2d30afb6bc

**Comment by TheIllusiveC4** [comment]
If pastebin isn't working, you could use other sites like gist or hastebin or any other alternative. I will need to see the crash log before I can begin to identify the cause. Also, could you elaborate more on your steps to reproduce? Detail it as if you were guiding me through your exact same steps. It's necessary so that I can try to reproduce it on my end exactly the way you did it. As it is now, I'm unsure when the crash occurs in relation to the error. 

One last thing, are you able to reproduce this reliably? If so, are you able to do it with only Polymorph and Silent Gear installed?

**Comment by ArMoReDfUrY** [comment]
Been trying on both haste and paste for an hour now. im getting fed up.

**Comment by ArMoReDfUrY** [comment]
waiting on gist upload now.

**Comment by ArMoReDfUrY** [comment]
log added.

**Comment by TheIllusiveC4** [comment]
Thank you. Can you address the other questions I had in my previous post? Mainly:

1) Could you elaborate more on your steps to reproduce this issue?
2) Are you able to reproduce this issue reliably? If so, are you able to do it with only Polymorph and Silent Gear installed?

**Comment by ArMoReDfUrY** [comment]
will do, need some time, but i'll get it tested later today.

**Comment by ArMoReDfUrY** [comment]
ok... testing done.

it seems polymorph does not like updates.

so i created a new world with just poly and silentgear. works great

i then created a new world on my pack setup and... works great.

so i removed poly and loaded my original world, tested, quit the game, then added poly back in and loaded in and... 
it works great!!

i had updated rats & silent gear just before this issue & they both had some recipe changes/additions.
thinking back i had a similar issue a few days ago and had pinned it on another mod... sigh false positives are a pain.

i think we need a new issue: "Polymorph dislikes mod updates on existing worlds 1.15.2"

**Comment by ArMoReDfUrY** [comment]
BTW the silent gear crafting station works fine with the collision button.

**Comment by TheIllusiveC4** [comment]
Do you remember which versions of Rats/Silent Gear you had before and after the issue occurred? If so, I'd like to know so I can test that aspect myself.

**Comment by ArMoReDfUrY** [comment]
you will need these library files, they stay in for both versions.

SilentLib-1.15.2-4.6.4+56
citadel-1.1.11

I started with these:

(Polymorph is installed)
rats-5.3.1-1.15.2
SilentGear-1.15.2-1.8.0+107
SilentMechanisms-1.15.2-0.7.2+52

make a world then update to these:

rats-5.3.2-1.15.2
SilentGear-1.15.2-1.8.1+108
SilentMechanisms-1.15.2-0.7.3+53

just know this might not be enough to do it, you may need more mods to inflate the recipe lists and trigger this bug.
hope not though, that will be a pain...

**Comment by ArMoReDfUrY** [comment]
just wanted to confirm that i have replicated this behaviour a dozen times at min.
had many updates the last few days and even had it happen after adding just Waddles mod (mod has no recipes at all just a loot table for the single entity)

and confirming that it can be resolved by loading the world once without polymorph installed.
you can then re-add polymorph for full functionality.

**Comment by TheIllusiveC4** [comment]
Sorry for the delay. Since I can't reproduce this, I can't verify if my attempted fix actually solves the issue. Just let me know if the issue still occurs with the 0.4 update.

**Comment by ArMoReDfUrY** [comment]
will check it out in a few hours. i gotta go do life for a bit :)

**Comment by mr-poyo** [comment]
Just wanted to post on my own personal experience. I have a custom modpack on my server and i recently updated Polymorph from 0.3 to 0.4 among some other mods. My crafting and even anvil use broke. It was failing to return recipes. Loaded a fresh single player world, worked fine. 

I have since reverted my server world back to polymorph 0.3 and the crafting/anvils are working properly again. 

Forge: 31.2.31

**Comment by TheIllusiveC4** [comment]
@mr-poyo Do the logs show any errors?

**Comment by mr-poyo** [comment]
No crash log, it was only an interaction problem. 
I have since reverted to 0.3, but I did look up some old zipped logs:

crash report: https://pastebin.com/G4Q5yKcS

If you need to me to dig more and grab a full log I can look. I snipped this one right after I logged in and did some craft testing. Hope it's useful.

**Comment by Vaelzan** [comment]
FYI - this appears to be a problem in 1.16.1 currently. Can't easily file a fresh issue since I'm on my phone but there's some additional information here: https://github.com/ValhelsiaTeam/Valhelsia/issues/221

**Comment by TheIllusiveC4** [comment]
@Vaelzan I don't see any logs or error messages mentioned in the linked issue. Can anyone provide either of those? If the recipes are failing, it should be logging something.

**Comment by Khytwel** [comment]
@TheIllusiveC4 Here's a log file sent on our issue.
[2020-08-05-2.log](https://github.com/TheIllusiveC4/Polymorph/files/5033896/2020-08-05-2.log)

**Comment by TheIllusiveC4** [comment]
@Khytwel Hmm, weird. So, there's nothing in the logs at all from Polymorph. That's not to say I don't think there's an issue here, as clearly it seems uninstalling Polymorph makes everything work again. But I'm unsure what's exactly happening. All Polymorph does is fetch the recipes, it shouldn't be doing anything to the recipes themselves.

I'll have to investigate further. Some additional questions: Does this only occur when updating worlds, or are fresh instances also cursed with broken recipes? Does this behavior apply to all types of crafting tables or only specific ones?

**Comment by Khytwel** [comment]
@TheIllusiveC4 For the new instances apparently no, but I'm not sure.

But I remember that versions of Forge above 32.0.70 that's where the bug is. I had updated the modpack once on a new version of Forge and that's when the bug appeared.  So we'd gone back to Forge 32.0.70 and there was no more problem.

Otherwise the bug only occurs on a few crafts like: raw basic processor from Refined Storage, applesauce from Pam's HarvestCraft, etc
![image](https://user-images.githubusercontent.com/54046029/89529678-69c59980-d7ed-11ea-978b-19e3f73e5246.png)
![image](https://user-images.githubusercontent.com/54046029/89529742-86fa6800-d7ed-11ea-942d-0a27355b1837.png)
![image](https://user-images.githubusercontent.com/54046029/89529783-94afed80-d7ed-11ea-9516-b1cf9fa20a3f.png)
![image](https://user-images.githubusercontent.com/54046029/89529797-9da0bf00-d7ed-11ea-8e33-1851048e0ad1.png)
![image](https://user-images.githubusercontent.com/54046029/89529822-aabdae00-d7ed-11ea-98a5-df818fcc4dde.png)
![image](https://user-images.githubusercontent.com/54046029/89529838-b4471600-d7ed-11ea-86ff-f93680ea31fa.png)

If you want you can find all our changelogs on our GitHub: https://github.com/ValhelsiaTeam/Valhelsia/blob/1.16/CHANGELOG.md
3.0.10 bug craft, 3.0.9 no bug craft, 3.0.8 no bug craft, 3.0.7 no bug craft, 3.0.6 bug craft, 3.0.5 no bug craft.

**Comment by rodrigokiller** [comment]
Hello fellow comrades
If any of you need more logs, screenshots or anything else, I can provide it for you. I can create a video too, even an TV connection to reproduce the error (sorry, I dont use containers for this).
You just need to mention me, I will be delighted to help

**Comment by Khytwel** [comment]
@TheIllusiveC4 Any news?

**Comment by TheIllusiveC4** [comment]
@Khytwel Sorry for the lack of updates. I am working on this, it's just a little involved because I want to refactor a core aspect of Polymorph to try and resolve this and all future related issues. This particular issue thread will remain closed as it is slightly different, but I've opened a new thread that you and anyone else can monitor for updates: https://github.com/TheIllusiveC4/Polymorph/issues/24
